server.py

```diff
@@ -0,0 +1,108 @@
+from sys import argv
+
+from http.server import BaseHTTPRequestHandler, HTTPServer
+from http.cookies import SimpleCookie
+from cgi import FieldStorage
+
+from jinja2 import Environment, FileSystemLoader
+
+
+HOST = '127.0.0.1'
+PORT = 8001
+
+if len(argv) > 1:
+    PORT = int(argv[1])
+
+
+class Handler(BaseHTTPRequestHandler):
+
+    def _get_cookies(self):
+        cookies = SimpleCookie(self.headers.get('Cookie'))
+
+        if len(cookies) > 0 and int(cookies['auth'].value) > 0:
+            return True
+
+        return False
+
+    def _get_template(self, page):
+        file_loader = FileSystemLoader('templates')
+        env = Environment(loader=file_loader)
+        return env.get_template(page)
+
+    def _set_cookies(self, value):
+        cookies = SimpleCookie()
+        cookies['auth'] = value
+        self.send_header('Set-Cookie', cookies.output(header='', sep=''))
+
+    def _charge(self):
+        pass
+
+    def _error(self):
+        pass
+
+    def _form(self):
+        pass
+
+    def _log_in(self):
+        self._set_cookies(100)
+
+    def _log_out(self):
+        self._set_cookies(0)
+
+    PAGES = {
```

Comment:
These look more like header handlers, rather than `PAGES`.

Also you use a different approach for the response content (while both headers and content are parts of the same response):

1. `smth = get_template()`
1. write `smth` into response within `do_GET/POST()`

You could do the same for headers as well:
1. `headers = get_headers()`
1. write headers into response within `do_GET/POST()` directly.

Reply:
I named this constant PAGES for readability purposes. For example let's take next code:
`if page not in self.PAGES.keys(): page = 'error.html'`
It's easier to understand what's going on when you try to find page in pages list, rather than searching for page in header handlers.

Reply:
Right. The point is the dict is usually perceived by its values (more than by keys).
So the values of your dict look more like header handlers (than some pages). 

Also, you use this dict for multiple purposes after all. You could have a more explicit (though more sophisticated) structure like: `PAGES = ({'name': 'charge', 'header_handler': _charge}, ...)` and then it'd be possible to use it more transparently for different use cases. 
But please pay (even more) attention to the second idea (`get_template()+get_headers()` as well).

Reply:
Any further actions here?

Reply:
I think this structure looks a bit less complicated and easier to use:
`TEMPLATES = { 'page1': (handler1_1, handler1_2), ... }`.

About template/headers thing. I'm a bit confused with this task. From what I understood you want me to write some header-formatted string in `_get_headers()` and then somehow put this string into response as headers. I understand how to do the first part, but have no idea about second, plus this looks like some sort of crutch to me. Correct me if I'm wrong.

Reply:
1. `PAGE/TEMPLATES` - the idea is that the collection/dict/.. (whatever it looks like after all) becomes more semantically correct while serving different goals.
1. `_get_headers()` - look, you write response text directly from `do_GET/POST()` while you write headers from nested methods. This doesn't look really natural because both text and headers are parts of the same response. It'd be better if both parts are written from the same scopes (either directly or from submethods). 

In both cases there's nothing wrong about the final result, but as your codebase grows (imagine it could grow at some point), this will bring more complications in code reading and refactoring.

Reply:
So would it be better if i write headers in _set_headers and text in _set_templates?

Reply:
any uniform approach would be better

Reply:
fixed